### DISCUSSION TITLE
[Dev Tools] zkup vs zk_toolbox vs zk_inception vs zk_supervisor vs zks: Why are there so many CLIs?

### DISCUSSION THREAD

**Discussion by ghost** [zkSync-Community-Hub/zksync-developers#680] [Dev Tooling] [answered]
### Environment

Testnet

### Select the Dev Tool you are using

ZKsync CLI

### Provide a brief description of the functionality you're trying to implement and the issue you are running into.

What's the difference between zkup, zk_toolbox, zk_inception, zk_supervisor and zks CLIs?
And why do you need so many different tools?

**Comment by alhenaai** [comment]
Thank you for your question! I’m unable to provide an answer at this moment, but I have forwarded your inquiry to our team. Someone will be with you shortly to assist further. Your patience is appreciated!

**Reply by bxpana** [reply, marked as answer]
- zkup: "a script designed to simplify the installation of [zk_toolbox](https://github.com/matter-labs/zksync-era/tree/main/zk_toolbox). It allows you to install the tool from a local directory or directly from a GitHub repository." [repo](https://github.com/matter-labs/zksync-era/tree/9821a20018c367ce246dba656daab5c2e7757973/zk_toolbox/zkup)
- zk_toolbox: "Toolkit for creating and managing ZK Stack chains." [repo](https://github.com/matter-labs/zksync-era/tree/9821a20018c367ce246dba656daab5c2e7757973/zk_toolbox#zk_toolbox)
- zk_inception: "ZK Inception facilitates the creation and management of an Elastic Chain ecosystem and ZK chains using ZK Stack." [docs](https://docs.zksync.io/zk-stack/running-a-zk-chain/quickstart#install-zk-inception)
- zk_supervisor: "Tools for developing ZKsync." [repo](https://github.com/matter-labs/zksync-era/tree/9821a20018c367ce246dba656daab5c2e7757973/zk_toolbox#zk-supervisor)
- zks: ~~not sure on this one but can double check.~~ [EDIT]: it's related to running the chain, but for developing ZKsync, testing, linting, working with db, some additional tools

The tools that you mentioned are all for building a ZK Chain using the ZK Stack. zk_toolbox is the tool that you would use with the other commands you listed.
